### PR TITLE
Added build cli options proposal

### PR DESCRIPTION
**Pull request by Roemer**
Added a simple proposal for the spec change in https://github.com/devcontainers/spec/pull/324

### CODE REVIEW

File: proposals/build-cli-options.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- https://github.com/devcontainers/spec/pull/324
+- https://github.com/microsoft/vscode-remote-release/issues/3545
+
+## Goal
+
+Extend the build section of the configuration with an option to pass additional arguments via cli.
+
+## Proposal
+
+Introduce the property `cliOptions` to the `buildOptions` in the schema. It is very similar to `runArgs`, which contains cli-arguments passed to the `run` command, the `cliOptions` passes cli-arguments to the `build` command.
```
Comment (samruddhikhandale) [inline proposals/build-cli-options.md:14]: I wonder if calling the new property as `options` makes sense. (eg `build.options`)

// cc @bamurtaugh @craiglpeters @Chuxel for product specific insights. Also, there are many other helpful suggestions from the author in the https://github.com/devcontainers/spec/pull/324 description. 

> build.buildArgs
build.cliArgs
build.cliOptions
build.options
build.commandArgs
build.commandOptions

File: proposals/build-cli-options.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
```
Comment (samruddhikhandale) [inline proposals/build-cli-options.md:15]: Can we also add a line in the proposal stating that the new property will be of type `Array (string)`

**Review by samruddhikhandale — COMMENTED** [review]
Thank you for taking the time in contributing this proposal, we appreciate it 👏

File: proposals/build-cli-options.md (hunk 1 above)
Comment (Roemer) [reply proposals/build-cli-options.md:14]: I am open for the naming. `options` seems a bit too generic for me. In the end, the field represents cli arguments passed to the command.

File: proposals/build-cli-options.md (hunk 2 above)
Comment (Roemer) [reply proposals/build-cli-options.md:15]: I adjusted the text a bit and also mention the type.

File: proposals/build-cli-options.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- https://github.com/devcontainers/spec/pull/324
+- https://github.com/microsoft/vscode-remote-release/issues/3545
+
+## Goal
+
+Extend the build section of the configuration with an option to pass additional arguments via cli.
+
+## Proposal
+
+Introduce the property `cliOptions` to the `buildOptions` in the schema as type `Array (string)`. This field contains elements that are directly passed as cli-arguments to the `build` command. It is similar to `runArgs` property which contains cli-arguments passed to the `run` command.
```
Comment (chrmarti) [inline proposals/build-cli-options.md:14]: Should this be "`cliArgs` to the `build` property" to match existing names?

File: proposals/build-cli-options.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+###  Execution
+
+When a dev container is built, all entries from the `cliOptions` are passed to the `build` command at the end.
```
Comment (chrmarti) [inline proposals/build-cli-options.md:18]: Change "at the end" to "after all other options" since the context path will still go last?

**Review by chrmarti — CHANGES_REQUESTED** [review]
Thanks for the proposal! Left a few questions.

File: proposals/build-cli-options.md (hunk 3 above)
Comment (Roemer) [reply proposals/build-cli-options.md]: Adjusted it, yes it's the `build` property (the type in the schema is `buildOptions`).

File: proposals/build-cli-options.md (hunk 4 above)
Comment (Roemer) [reply proposals/build-cli-options.md]: Good catch. True of course, adjusted it.

File: proposals/build-cli-options.md (hunk 1 above)
Comment (bamurtaugh) [reply proposals/build-cli-options.md]: Thanks for providing all the great suggestions! I lean toward `build.options` too - it feels the cleanest and most concise. While other areas may also have things called "options," I don't anticipate us wanting to refer to other things as `build.options`.

**Comment by Roemer** [comment]
I just stumpled uppon problems with features as well. Should this proposal also add a point that those `options` should also be added when adding features?

File: proposals/build-cli-options.md (hunk 1 above)
Comment (Roemer) [reply proposals/build-cli-options.md]: I renamed it to `options`.

**Comment by chrmarti** [comment]
The image is built in a single `docker build` run, adding build args to features might allow for conflicting arguments.

**Review by chrmarti — APPROVED** [review]
LGTM!

File: proposals/build-options.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    "options": [
+      "--add-host=host.docker.internal:host-gateway"
+    ]
+  },
+}
+```
+
+### Use host network
+
+The following example allows to sue the host network while building an image:
```
Comment (eljog) [inline proposals/build-options.md:39]: typo?
```suggestion
The following example allows to use the host network while building an image:
```

**Review by eljog — APPROVED** [review]
⚡

**Review by 113ll3 — COMMENTED** [review]
Looks like a buzz

**Pull request merged**